### ISSUE TITLE
write_prometheus fails to initialize during service restart

### ISSUE THREAD

**Issue by twblamer** [collectd/collectd#2715]
*   Version of collectd: 5.8.0
*   Operating system / distribution: RHEL 7, Fedora 25 (haven't tested others)

This looks like a regression in 5.8.0. I don't see the behavior in 5.7.2.

## Expected behavior

The write_prometheus plugin should start normally when I run "systemctl restart collectd".

## Actual behavior

The write_prometheus plugin fails when I restart the collectd service:

```
Mar 16 16:29:06 localhost collectd[655]: write_prometheus plugin: Opening a listening socket failed.
Mar 16 16:29:06 localhost collectd[655]: write_prometheus plugin: MHD_start_daemon() failed.
Mar 16 16:29:06 localhost collectd[655]: Initialization of plugin `write_prometheus' failed with status -1. Plugin will be unloade
Mar 16 16:29:06 localhost collectd[655]: plugin_unregister_read: No such read function: write_prometheus
Mar 16 16:29:06 localhost collectd[655]: Error: one or more plugin init callbacks failed.
```

If I explicitly stop the service, then sleep for 100 seconds, then start the service, I avoid the problem. However, this is a real pain when I'm working on a plugin and trying to test my changes.
```
systemctl stop collectd
sleep 100
systemctl start collectd
...
Mar 16 16:33:14 localhost collectd[2033]: plugin_load: plugin "write_prometheus" successfully loaded.
...
Mar 16 16:33:14 localhost collectd[2033]: Warning: MHD_USE_THREAD_PER_CONNECTION must be used only with MHD_USE_INTERNAL_POLLING_THREAD. Flag MHD_USE_INTERNAL_POLLING_THREAD was added. Consider setting MHD_USE_INTERNAL_POLLING_THREAD explicitly.
```
## Steps to reproduce

On Fedora 25 (should be similar for other distros):
```bash
sudo dnf install collectd-5.8.0 collectd-write_prometheus-5.8.0
sudo sed -i 's/^#LoadPlugin write_prometheus/LoadPlugin write_prometheus/' /etc/collectd.conf
sudo systemctl start collectd
sudo systemctl restart collectd
```

**Comment by twblamer** [comment]
Looks like this got fixed in 2c78bdf94c0fdd6240d5d202cf5aba12ac8c5a20, any thoughts on getting this into a release?

**Comment by petarkozic** [comment]
Also, I have same problem on Debian 8 Jessie.

collectd 5.8.0.63.ga68a482

**Comment by rpv-tomsk** [comment]
Hi!

We need more info about your system, used Collectd configuration.
Also we need more debug output and, maybe, your patches for this to be fixed.

**Comment by rpv-tomsk** [comment]
>collectd 5.8.0.63.ga68a482
>Collectd version: collectd 5.8.0.70.gbc38149

I'm unable to find these commits/source. Where you got them from?

**Comment by rpv-tomsk** [comment]
>It seems that it is still failing to start on boot on 5.8.0 + 2c78bdf on Arch Linux.

We need more info about your system: used Collectd configuration (is `Port` directive was used?), libmicrohttpd version. Also we need logs.

**Comment by petarkozic** [comment]
@rpv-tomsk 
Thank you very much on your answer.
Please do you know when will this commit in repository for Debian ?

**Comment by rpv-tomsk** [comment]
I don't know when issue #2737 will be resolved. It does not depends on me, unfortunately.

**Comment by mrunge** [comment]
I still have this issue with cherry-picked patch https://github.com/collectd/collectd/pull/2737/commits/2c78bdf94c0fdd6240d5d202cf5aba12ac8c5a20

latest libmicrohttpd from here: https://ftp.gnu.org/gnu/libmicrohttpd/ built for CentOS 7 http://cbs.centos.org/koji/buildinfo?buildID=22621

There must be something else missing (in my case).

**Comment by rpv-tomsk** [comment]
>What happens if you just build and run master? I did not cherry pick 2c78bdf, I just built master (whatever 
HEAD was on 2018-03-16) and that seemed to fix the issue for me.

Just now looked: this patch exists in collectd-5.7 and master branches and is missing only in collectd-5.8.

**Comment by rpv-tomsk** [comment]
Patch was merged to collectd-5.8.
